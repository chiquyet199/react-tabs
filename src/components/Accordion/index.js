import Accordion from "./Accordion";
import AccordionWithCount from "./AccordionWithCount";
import AccordionWithIconTitle from "./AccordionWithIconTitle";

export {Accordion, AccordionWithCount, AccordionWithIconTitle};
