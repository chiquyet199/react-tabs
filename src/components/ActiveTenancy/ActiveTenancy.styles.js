import styledNative from "styled-components/native";
import {themed} from "../../utils/common";
import Icon from "../Icon";

export const ThemedText = themed(styledNative.Text`
  color: ${({globaltheme, theme}) => theme[globaltheme].textColor};
`);
export const HorizontalStack = styledNative.View`
  flex-direction: row;
`;
export const SectionTitle = styledNative(ThemedText)`
  font-size: 24;
`;
export const SectionText = styledNative(ThemedText)`
  font-size: 18;
`;
export const SectionTextSmall = styledNative(ThemedText)`
  font-size: 16;
`;
export const Section = themed(styledNative.View`
  padding: 11px 0 13px;
  margin: 0 17px 0 18px;
  border-bottom-color: ${({globaltheme, theme}) =>
    theme[globaltheme].boxBorder};
  border-bottom-width: 1;
`);
export const ThemedIcon = themed(styledNative(Icon)`
  color: ${({globaltheme, theme}) => theme[globaltheme].textColor};
`);
