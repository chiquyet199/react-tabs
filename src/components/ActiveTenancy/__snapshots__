/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/ActiveTenancy/__native_snapshots__