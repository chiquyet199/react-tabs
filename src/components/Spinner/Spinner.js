import React from "react";
import {string, bool, oneOf} from "prop-types";
import NativeSpinner from "./NativeSpinner";
import {themed} from "../../utils/common";

const ANIMATION = ["none", "slide", "fade"];
const SIZES = ["small", "normal", "large"];

export const Spinner = props => <NativeSpinner {...props} />;

Spinner.propTypes = {
  visible: bool,
  cancelable: bool,
  textContent: string,
  color: string,
  animation: oneOf(ANIMATION),
  size: oneOf(SIZES),
  overlayColor: string,
};

export default themed(Spinner);
