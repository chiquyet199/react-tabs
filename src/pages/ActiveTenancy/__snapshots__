/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/pages/ActiveTenancy/__native_snapshots__