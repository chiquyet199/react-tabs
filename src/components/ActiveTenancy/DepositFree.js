import React, {Fragment} from "react";
import {connect} from "react-redux";
import {shape, string} from "prop-types";
import styledNative from "styled-components/native";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import {Accordion, AccordionWithIconTitle} from "../Accordion";
import {
  HorizontalStack,
  Section,
  SectionTitle,
  SectionText,
  ThemedIcon,
} from "./ActiveTenancy.styles";
import {keyFromString} from "../../utils/string";

const DepositDocumentDetails = styledNative.View`
  flex-grow: 1;
`;

export const DepositFree = ({lease, translations}) => (
  <Accordion
    title={({collapsed}) => (
      <AccordionWithIconTitle
        headerIconName="claims"
        title={translations.title}
        subTitle={translations.subTitle}
        collapsed={collapsed}
      />
    )}
  >
    <Fragment>
      {lease.documents.map(document => (
        <Section key={keyFromString(document.type)}>
          <HorizontalStack>
            <DepositDocumentDetails>
              <SectionTitle>{translations.title}</SectionTitle>
              <SectionText>
                {translations.documentTypes[document.type]}
              </SectionText>
            </DepositDocumentDetails>
            <ThemedIcon name="legal" size={32} />
          </HorizontalStack>
        </Section>
      ))}
    </Fragment>
  </Accordion>
);

DepositFree.propTypes = {
  lease: LeaseSpec,
  translations: shape({
    subTitle: string.isRequired,
    title: string.isRequired,
    documentTypes: shape({
      POLICY_SCHEDULE: string.isRequired,
      STATEMENT_OF_FACTS: string.isRequired,
      POLICY_WORDING: string.isRequired,
      INSURANCE_PRODUCT_INFORMATION_DOCUMENT: string.isRequired,
      INSURANCE_SERVICE_INFORMATION_DOCUMENT: string.isRequired,
      AST: string.isRequired,
    }),
  }),
};

const mapStateToProps = state => ({
  translations: state.locale.translations.common.depositFree,
});

export default connect(mapStateToProps)(DepositFree);
