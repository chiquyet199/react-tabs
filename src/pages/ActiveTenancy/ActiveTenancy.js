import React, {PureComponent, Fragment} from "react";
import {connect} from "react-redux";
import {func, shape, string} from "prop-types";
import {
  DepositFree,
  Housemates,
  TenancyTerms,
} from "../../components/ActiveTenancy";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import HeaderPropertySummary from "../../components/HeaderPropertySummary";
import Spinner from "../../components/Spinner";
import {findLeaseByProperty} from "../../actions/leases";
import {showTopbarCenterText} from "../../actions/topbar";

export class ActiveTenancy extends PureComponent {
  componentDidMount() {
    const {
      findLeaseByProperty: fetchLease,
      match: {
        params: {id},
      },
    } = this.props;

    fetchLease(id);
  }

  render() {
    const {lease, showTopbarCenterText: setTopbarText} = this.props;

    if (lease) {
      setTopbarText(lease.property.address.line1);
    }

    return (
      <Fragment>
        <Spinner visible={lease === undefined} />
        {lease && (
          <Fragment>
            <HeaderPropertySummary lease={lease} />
            <TenancyTerms lease={lease} />
            {lease.documents.length && <DepositFree lease={lease} />}
            {lease.renters.length > 1 && <Housemates lease={lease} />}
          </Fragment>
        )}
      </Fragment>
    );
  }
}

ActiveTenancy.propTypes = {
  findLeaseByProperty: func.isRequired,
  lease: LeaseSpec,
  match: shape({
    params: shape({
      id: string.isRequired,
    }).isRequired,
  }).isRequired,
  showTopbarCenterText: func.isRequired,
};

const mapsStateToProps = state => ({
  lease: state.property && state.property.lease,
});

const mapDispatchToProps = {
  findLeaseByProperty,
  showTopbarCenterText,
};

export default connect(
  mapsStateToProps,
  mapDispatchToProps,
)(ActiveTenancy);
