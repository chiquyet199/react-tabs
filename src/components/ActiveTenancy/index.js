import DepositFree from "./DepositFree";
import Housemates from "./Housemates";
import TenancyTerms from "./TenancyTerms";

export {DepositFree, Housemates, TenancyTerms};
