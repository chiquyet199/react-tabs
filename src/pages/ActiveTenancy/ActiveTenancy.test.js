import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";
import {ActiveTenancy} from "./ActiveTenancy";
import {DepositFree, Housemates} from "../../components/ActiveTenancy";
import Spinner from "../../components/Spinner";

describe("ActiveTenancy page tests", () => {
  it(`matches snapshot on ${Platform.OS}`, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it("sets spinner isLoading to true when lease not provided", () => {
    expect(getComponent({lease: undefined}).find(Spinner)).toHaveProp(
      "visible",
      true,
    );
  });

  it("sets topbar text to be first line of property address", () => {
    const property = getMockProperty();
    property.address = {
      line1: "10 Downing Street",
      town: "London",
      postCode: "SW1A 2AA",
    };
    const showTopbarCenterText = jest.fn();
    getComponent({lease: getMockLease(property), showTopbarCenterText});
    expect(showTopbarCenterText).toHaveBeenCalledWith(property.address.line1);
  });

  it("doesn't set topbar text when lease not provided", () => {
    const showTopbarCenterText = jest.fn();
    getComponent({lease: undefined, showTopbarCenterText});
    expect(showTopbarCenterText).not.toHaveBeenCalled();
  });

  it("omits deposit free section if lease has no documents", () => {
    const lease = getMockLease();
    lease.documents = [];
    const component = getComponent({lease});
    expect(component.find(DepositFree)).not.toExist();
  });

  it("doesn't display housemates section if there's only one tenant", () => {
    const lease = getMockLease();
    lease.renters.length = 1;
    const component = getComponent({lease});
    expect(component.find(Housemates)).not.toExist();
  });

  it("requests lease by property id on mount", () => {
    const findLeaseByProperty = jest.fn();
    const match = {params: {id: "abc-123"}};
    getComponent({findLeaseByProperty, match});
    expect(findLeaseByProperty).toHaveBeenCalledWith(match.params.id);
  });

  const getComponent = props => {
    const parsedProps = {
      findLeaseByProperty: () => {},
      lease: getMockLease(),
      match: {
        params: {
          id: "",
        },
      },
      showTopbarCenterText: () => {},
      ...props,
    };

    return shallow(<ActiveTenancy {...parsedProps} />);
  };
});

const getMockLease = property => ({
  id: "abc-123",
  branchId: "def-456",
  property: property || getMockProperty(),
  renters: [
    {
      userId: "ghi-789",
      depositOptions: [{type: "DEPOSIT_FREE"}],
      firstName: "Bill",
      lastName: "Oddy",
    },
    {
      userId: "ghi-987",
      depositOptions: [{type: "DEPOSIT_FREE"}],
      firstName: "Damon",
      lastName: "Albarn",
    },
  ],
  status: "ACTIVE",
  rentAmount: 1234,
  rentFrequency: "MONTHLY",
  duration: {
    minimumMonths: 6,
    startDate: "2018-11-06T00:00:00",
  },
  conditionsFreeText: "",
  documents: [
    {
      type: "POLICY_SCHEDULE",
      ref: "ghi-998",
      createdAt: "2018-06-01T15:32:48",
    },
    {
      type: "STATEMENT_OF_FACTS",
      ref: "ghi-712",
      createdAt: "2018-06-01T15:48:48",
    },
  ],
});

const getMockProperty = () => ({
  id: "abc-123",
  presentation: {
    bathroomsCount: 1,
    bedroomsCount: 2,
    furnishedType: "PART_FURNISHED",
    hasOutsideSpace: true,
    hasParking: true,
    petsAllowed: false,
    propertyType: "HOUSE",
    smokingAllowed: false,
  },
  rentFrequency: "WEEKLY",
  rentAmount: 1234.99,
  depositAmount: 1000,
  depositFreeAvailable: true,
  propertyImages: [
    {
      id: "def-456",
      url: "http://www.findyourcanopy.com/img123.jpeg",
      isPrimary: false,
    },
    {
      id: "def-457",
      url:
        "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
      isPrimary: true,
    },
    {
      id: "def-458",
      url: "http://www.findyourcanopy.com/img125.jpeg",
      isPrimary: false,
    },
  ],
  landlord: {},
  address: {
    line1: "29 Acacia Road",
    town: "Beanotown",
    state: "Aberdeenshire",
    postCode: "BE1 1BE",
    countryCode: "UK",
  },
  createdBy: "Homogenous cohorts",
  agencyId: "22",
  branchId: "44",
});
