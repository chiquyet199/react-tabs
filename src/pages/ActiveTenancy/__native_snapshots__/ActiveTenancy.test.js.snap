// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveTenancy page tests matches snapshot on ios 1`] = `
<Fragment>
  <Connect(WithTheme(Spinner))
    visible={false}
  />
  <Connect(HeaderPropertySummary)
    lease={
      Object {
        "branchId": "def-456",
        "conditionsFreeText": "",
        "documents": Array [
          Object {
            "createdAt": "2018-06-01T15:32:48",
            "ref": "ghi-998",
            "type": "POLICY_SCHEDULE",
          },
          Object {
            "createdAt": "2018-06-01T15:48:48",
            "ref": "ghi-712",
            "type": "STATEMENT_OF_FACTS",
          },
        ],
        "duration": Object {
          "minimumMonths": 6,
          "startDate": "2018-11-06T00:00:00",
        },
        "id": "abc-123",
        "property": Object {
          "address": Object {
            "countryCode": "UK",
            "line1": "29 Acacia Road",
            "postCode": "BE1 1BE",
            "state": "Aberdeenshire",
            "town": "Beanotown",
          },
          "agencyId": "22",
          "branchId": "44",
          "createdBy": "Homogenous cohorts",
          "depositAmount": 1000,
          "depositFreeAvailable": true,
          "id": "abc-123",
          "landlord": Object {},
          "presentation": Object {
            "bathroomsCount": 1,
            "bedroomsCount": 2,
            "furnishedType": "PART_FURNISHED",
            "hasOutsideSpace": true,
            "hasParking": true,
            "petsAllowed": false,
            "propertyType": "HOUSE",
            "smokingAllowed": false,
          },
          "propertyImages": Array [
            Object {
              "id": "def-456",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img123.jpeg",
            },
            Object {
              "id": "def-457",
              "isPrimary": true,
              "url": "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
            },
            Object {
              "id": "def-458",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img125.jpeg",
            },
          ],
          "rentAmount": 1234.99,
          "rentFrequency": "WEEKLY",
        },
        "rentAmount": 1234,
        "rentFrequency": "MONTHLY",
        "renters": Array [
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Bill",
            "lastName": "Oddy",
            "userId": "ghi-789",
          },
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Damon",
            "lastName": "Albarn",
            "userId": "ghi-987",
          },
        ],
        "status": "ACTIVE",
      }
    }
  />
  <Connect(TenancyTerms)
    lease={
      Object {
        "branchId": "def-456",
        "conditionsFreeText": "",
        "documents": Array [
          Object {
            "createdAt": "2018-06-01T15:32:48",
            "ref": "ghi-998",
            "type": "POLICY_SCHEDULE",
          },
          Object {
            "createdAt": "2018-06-01T15:48:48",
            "ref": "ghi-712",
            "type": "STATEMENT_OF_FACTS",
          },
        ],
        "duration": Object {
          "minimumMonths": 6,
          "startDate": "2018-11-06T00:00:00",
        },
        "id": "abc-123",
        "property": Object {
          "address": Object {
            "countryCode": "UK",
            "line1": "29 Acacia Road",
            "postCode": "BE1 1BE",
            "state": "Aberdeenshire",
            "town": "Beanotown",
          },
          "agencyId": "22",
          "branchId": "44",
          "createdBy": "Homogenous cohorts",
          "depositAmount": 1000,
          "depositFreeAvailable": true,
          "id": "abc-123",
          "landlord": Object {},
          "presentation": Object {
            "bathroomsCount": 1,
            "bedroomsCount": 2,
            "furnishedType": "PART_FURNISHED",
            "hasOutsideSpace": true,
            "hasParking": true,
            "petsAllowed": false,
            "propertyType": "HOUSE",
            "smokingAllowed": false,
          },
          "propertyImages": Array [
            Object {
              "id": "def-456",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img123.jpeg",
            },
            Object {
              "id": "def-457",
              "isPrimary": true,
              "url": "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
            },
            Object {
              "id": "def-458",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img125.jpeg",
            },
          ],
          "rentAmount": 1234.99,
          "rentFrequency": "WEEKLY",
        },
        "rentAmount": 1234,
        "rentFrequency": "MONTHLY",
        "renters": Array [
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Bill",
            "lastName": "Oddy",
            "userId": "ghi-789",
          },
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Damon",
            "lastName": "Albarn",
            "userId": "ghi-987",
          },
        ],
        "status": "ACTIVE",
      }
    }
  />
  <Connect(DepositFree)
    lease={
      Object {
        "branchId": "def-456",
        "conditionsFreeText": "",
        "documents": Array [
          Object {
            "createdAt": "2018-06-01T15:32:48",
            "ref": "ghi-998",
            "type": "POLICY_SCHEDULE",
          },
          Object {
            "createdAt": "2018-06-01T15:48:48",
            "ref": "ghi-712",
            "type": "STATEMENT_OF_FACTS",
          },
        ],
        "duration": Object {
          "minimumMonths": 6,
          "startDate": "2018-11-06T00:00:00",
        },
        "id": "abc-123",
        "property": Object {
          "address": Object {
            "countryCode": "UK",
            "line1": "29 Acacia Road",
            "postCode": "BE1 1BE",
            "state": "Aberdeenshire",
            "town": "Beanotown",
          },
          "agencyId": "22",
          "branchId": "44",
          "createdBy": "Homogenous cohorts",
          "depositAmount": 1000,
          "depositFreeAvailable": true,
          "id": "abc-123",
          "landlord": Object {},
          "presentation": Object {
            "bathroomsCount": 1,
            "bedroomsCount": 2,
            "furnishedType": "PART_FURNISHED",
            "hasOutsideSpace": true,
            "hasParking": true,
            "petsAllowed": false,
            "propertyType": "HOUSE",
            "smokingAllowed": false,
          },
          "propertyImages": Array [
            Object {
              "id": "def-456",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img123.jpeg",
            },
            Object {
              "id": "def-457",
              "isPrimary": true,
              "url": "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
            },
            Object {
              "id": "def-458",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img125.jpeg",
            },
          ],
          "rentAmount": 1234.99,
          "rentFrequency": "WEEKLY",
        },
        "rentAmount": 1234,
        "rentFrequency": "MONTHLY",
        "renters": Array [
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Bill",
            "lastName": "Oddy",
            "userId": "ghi-789",
          },
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Damon",
            "lastName": "Albarn",
            "userId": "ghi-987",
          },
        ],
        "status": "ACTIVE",
      }
    }
  />
  <Connect(Housemates)
    lease={
      Object {
        "branchId": "def-456",
        "conditionsFreeText": "",
        "documents": Array [
          Object {
            "createdAt": "2018-06-01T15:32:48",
            "ref": "ghi-998",
            "type": "POLICY_SCHEDULE",
          },
          Object {
            "createdAt": "2018-06-01T15:48:48",
            "ref": "ghi-712",
            "type": "STATEMENT_OF_FACTS",
          },
        ],
        "duration": Object {
          "minimumMonths": 6,
          "startDate": "2018-11-06T00:00:00",
        },
        "id": "abc-123",
        "property": Object {
          "address": Object {
            "countryCode": "UK",
            "line1": "29 Acacia Road",
            "postCode": "BE1 1BE",
            "state": "Aberdeenshire",
            "town": "Beanotown",
          },
          "agencyId": "22",
          "branchId": "44",
          "createdBy": "Homogenous cohorts",
          "depositAmount": 1000,
          "depositFreeAvailable": true,
          "id": "abc-123",
          "landlord": Object {},
          "presentation": Object {
            "bathroomsCount": 1,
            "bedroomsCount": 2,
            "furnishedType": "PART_FURNISHED",
            "hasOutsideSpace": true,
            "hasParking": true,
            "petsAllowed": false,
            "propertyType": "HOUSE",
            "smokingAllowed": false,
          },
          "propertyImages": Array [
            Object {
              "id": "def-456",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img123.jpeg",
            },
            Object {
              "id": "def-457",
              "isPrimary": true,
              "url": "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
            },
            Object {
              "id": "def-458",
              "isPrimary": false,
              "url": "http://www.findyourcanopy.com/img125.jpeg",
            },
          ],
          "rentAmount": 1234.99,
          "rentFrequency": "WEEKLY",
        },
        "rentAmount": 1234,
        "rentFrequency": "MONTHLY",
        "renters": Array [
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Bill",
            "lastName": "Oddy",
            "userId": "ghi-789",
          },
          Object {
            "depositOptions": Array [
              Object {
                "type": "DEPOSIT_FREE",
              },
            ],
            "firstName": "Damon",
            "lastName": "Albarn",
            "userId": "ghi-987",
          },
        ],
        "status": "ACTIVE",
      }
    }
  />
</Fragment>
`;
